server.cc

```diff
@@ -126,7 +126,6 @@ RepStatus CheckpointProducer::GetStatus(
   // We expect to get the same checkpoint_id as we provided in the CreateCheckpoint call
   if (req.checkpoint_id != checkpoint_id_) {
     logger->Log(Severity::ERROR, FormatString("Invalid checkpoint id\n"));
-    DestroyCheckpoint();
     return RepStatus(Code::DB_FAILURE, Severity::ERROR, FormatString("Invalid checkpoint id\n"));
```

Comment:
What about other DestroyCheckpoint in this function? Can you remove them?

Reply:
yes